pkgs/development/embedded/arduino/arduinoOTA/default.nix

```diff
@@ -0,0 +1,32 @@
+{ lib
+, buildGoModule
+, fetchFromGitHub
+, nix-update-script
+}:
+buildGoModule rec {
+  pname = "arduinoOTA";
+  version = "1.4.1";
+
+  src = fetchFromGitHub {
+    owner = "arduino";
+    repo = pname;
+    rev = "${version}";
+    hash = "sha256-HaNMkeV/PDEotYp8+rUKFaBxGbZO8qA99Yp2sa6glz8=";
+  };
+
+  vendorHash = null;
+
+  postPatch = ''
+    sed -i 's/\(versionString        = "\)/\1${version}/' version/version.go
```

Comment:
In nixpkgs there is the [`substituteInPlace`](https://nixos.org/manual/nixpkgs/stable/#fun-substituteInPlace) utility (for examples grep through nixpkgs) which is preferred over sed.